src/hash_table.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+from typing import Union
+import random
+
+
+class Node:
+    def __init__(self, key: str, val: Union[str, float], next: Node = None):
+        self.key = key
+        self.val = val
+        self.next = next
+
+
+class ChainHashTable:
+    def __init__(self, n: int):
+        self.__bucket = [Node(None, None)] * n
+
+    def insert(self, key: str, val: Union[str, float]) -> bool:
+        node: Node = self.__bucket[self._hash(key)]
+        while node.next:
+            node = node.next
+            if node.key == key:
+                node.val = val
+                return True
+        node.next = Node(key, val)
+        return True
+
+    def remove(self, key: str) -> bool:
+        node: Node = self.__bucket[self._hash(key)]
+        while node and node.next:
```

Comment:
I dont think we need to check node.next
This while never reach to the last element, so the last element never be removed.